Tests/DocumentValueResolverFunctionalTest.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Bundle\MongoDBBundle\Tests;
+
+use Doctrine\Bundle\MongoDBBundle\DoctrineMongoDBBundle;
+use Doctrine\Bundle\MongoDBBundle\Tests\Fixtures\FooBundle\Controller\DocumentValueResolverController;
+use Doctrine\Bundle\MongoDBBundle\Tests\Fixtures\FooBundle\Document\User;
+use Doctrine\Bundle\MongoDBBundle\Tests\Fixtures\FooBundle\FooBundle;
+use Doctrine\ODM\MongoDB\DocumentManager;
+use Symfony\Bundle\FrameworkBundle\FrameworkBundle;
+use Symfony\Bundle\FrameworkBundle\Kernel\MicroKernelTrait;
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+use Symfony\Component\Config\Loader\LoaderInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\HttpKernel\Kernel;
+use Symfony\Component\Routing\Loader\Configurator\RoutingConfigurator;
+
+use function sys_get_temp_dir;
+use function uniqid;
+
+/**
+ * @requires PHP 8.0
+ * @requires function \Symfony\Bridge\Doctrine\Attribute\MapEntity::__construct
```

Comment:
oh this is cool, didn't know one can use `@requires` this way!

Reply:
https://docs.phpunit.de/en/9.6/incomplete-and-skipped-tests.html#incomplete-and-skipped-tests-requires-tables-api yeah, never used before to be honest 😬